### PR TITLE
Add usaco 1037 cpp solution

### PR DESCRIPTION
**Pull request by RmdanJr**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
indent with tabs & use camelcase

File: solutions/usaco-1037.mdx [hunk 1]
```diff
@@ -14,6 +14,87 @@ By simulating handshakes for all values of $K$, we can find the minimum, maximum
...
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+using vi = vector<int>;
+#define all(x) begin(x), end(x)
+#define sz(x) (int)(x).size()
+
+void setIO(string name = "") {
+  ios_base::sync_with_stdio(0);
+  cin.tie(0);
```
[the remark this answers is not in the document]
Comment (RmdanJr) [reply solutions/usaco-1037.mdx:29]: I've used usaco ide to convert spaces to tabs. Sorry, I'll update it now.

[the remark this answers is not in the document]
Comment (RmdanJr) [reply solutions/usaco-1037.mdx]: I think the code is correctly indented now.

File: solutions/usaco-1037.mdx [hunk 2]
```diff
@@ -14,6 +14,88 @@ By simulating handshakes for all values of $K$, we can find the minimum, maximum
...
+void setIO(string name = "") {
+	ios_base::sync_with_stdio(0);
+	cin.tie(0);
+	if (sz(name)) {
+		freopen((name + ".in").c_str(), "r", stdin);
+		freopen((name + ".out").c_str(), "w", stdout);
+	}
+}
+
+struct shake {
```
Comment (SansPapyrus683) [inline solutions/usaco-1037.mdx:36]: rename to `Shake`

File: solutions/usaco-1037.mdx [hunk 3]
```diff
@@ -14,6 +14,88 @@ By simulating handshakes for all values of $K$, we can find the minimum, maximum
...
+	int t, x, y;
+	bool operator<(shake tmp) { return t < tmp.t; }
```
Comment (SansPapyrus683) [inline solutions/usaco-1037.mdx:38]:
```suggestion
	bool operator<(const shake& tmp) { return t < tmp.t; }
```

File: solutions/usaco-1037.mdx [hunk 4]
```diff
@@ -14,6 +14,88 @@ By simulating handshakes for all values of $K$, we can find the minimum, maximum
...
+	setIO("tracing");
+	int n, t;
+	string s;
+	cin >> n >> t >> s;
+	vector<shake> shakes(t);
+	vector<bool> cowsInfected(n);
+	unordered_set<int> possiblePatientZero;
+	int minK = t, maxK = -1;
+	// checks if pair {patientZero, k} is consistent
+	// with handshakes data over time
```
Comment (SansPapyrus683) [inline solutions/usaco-1037.mdx:51]: put this in one comment

File: solutions/usaco-1037.mdx [hunk 5]
```diff
@@ -14,6 +14,88 @@ By simulating handshakes for all values of $K$, we can find the minimum, maximum
...
+	auto isConsistent = [&](int patientZero, int k) {
+		vector<bool> tmpInfected(n);
+		vector<int> numShakes(n);
+		tmpInfected[patientZero] = true;
+		for (shake sh : shakes) {
+			if (tmpInfected[sh.x])
```
Comment (SansPapyrus683) [inline solutions/usaco-1037.mdx:57]: braces

File: solutions/usaco-1037.mdx [hunk 6]
```diff
@@ -14,6 +14,88 @@ By simulating handshakes for all values of $K$, we can find the minimum, maximum
...
+				tmpInfected[sh.y] = true;
+			if (numShakes[sh.y] <= k && tmpInfected[sh.y])
+				tmpInfected[sh.x] = true;
+		}
+		for (int i = 0; i < n; i++)
+			if (tmpInfected[i] != cowsInfected[i])
+				return false;
+		return true;
+	};
+	// marks infected cows
```
Comment (SansPapyrus683) [inline solutions/usaco-1037.mdx:71]: separate sections of code with whitespace

File: solutions/usaco-1037.mdx [hunk 7]
```diff
@@ -14,6 +14,88 @@ By simulating handshakes for all values of $K$, we can find the minimum, maximum
...
+			if (isConsistent(patientZero, k)) {
+				possiblePatientZero.insert(patientZero);
+				minK = min(minK, k);
+				maxK = max(maxK, k);
+			}
+		}
+	}
+	cout << possiblePatientZero.size() << " " << minK << " "
+		<< (maxK == t ? "Infinity" : to_string(maxK)) << endl;
+	return 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-1037.mdx:93]:
```suggestion
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1037.mdx [hunk 8]
```diff
@@ -14,6 +14,99 @@ By simulating handshakes for all values of $K$, we can find the minimum, maximum
...
+
+int main() {
+	setIO("tracing");
+
+	int n, t;
+	string s;
+	cin >> n >> t >> s;
+
+	vector<Shake> shakes(t);
+	vector<bool> cowsInfected(n);
```
Comment (SansPapyrus683) [inline solutions/usaco-1037.mdx:49]: use `snake_case` to name variables

Comment (RmdanJr) [reply solutions/usaco-1037.mdx:49]: Ok, I will. But you told me in the conversation above to use camelCase.

Comment (RmdanJr) [reply solutions/usaco-1037.mdx]: Updated

File: solutions/usaco-1037.mdx [hunk 9]
```diff
@@ -14,6 +14,99 @@ By simulating handshakes for all values of $K$, we can find the minimum, maximum
...
 
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+using vi = vector<int>;
+#define all(x) begin(x), end(x)
+#define sz(x) (int)(x).size()
```
Comment (SansPapyrus683) [inline solutions/usaco-1037.mdx:25]: can you detemplify the code plz? (optional)

Comment (RmdanJr) [reply solutions/usaco-1037.mdx]: done!

**Review by SansPapyrus683 — APPROVED** [review]
nice
@envyaims

**Review by envyaims — APPROVED** [review]
nice

File: solutions/usaco-1037.mdx [hunk 10]
```diff
@@ -14,6 +14,95 @@ By simulating handshakes for all values of $K$, we can find the minimum, maximum
...
+		for (int i = 0; i < n; i++) {
+			if (tmp_infected[i] != cows_infected[i]) {
+				return false;
+			}
+		}
+		return true;
+	};
+
+	// marks infected cows
+	for (int x = 0; x < n; x++) {
```
Comment (envyaims) [inline solutions/usaco-1037.mdx:77]: why `x`?

Comment (RmdanJr) [reply solutions/usaco-1037.mdx:77]: Just a variable to loop through all chars of s;

Comment (RmdanJr) [reply solutions/usaco-1037.mdx:77]: Any thoughts about it?

Comment (envyaims) [reply solutions/usaco-1037.mdx:77]: why not just `i` lol

Comment (RmdanJr) [reply solutions/usaco-1037.mdx:77]: I think naming it `i` or `x` wouldn't make a difference. Just an iterator.

Comment (envyaims) [reply solutions/usaco-1037.mdx:77]: but `i` is more standard 🤔 
if theres no real reason id recommend just changing it back

Comment (RmdanJr) [reply solutions/usaco-1037.mdx]: You are right. I updated it.

**Pull request merged**